### PR TITLE
CONT-1219 : fail ci for puppetlabs members if no label

### PR DESCRIPTION
**Pull request by malikparvez**
Adding the support for PR's to fail my puppetlabs members if no label is added manually

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -25,7 +26,13 @@ export async function run(): Promise<void> {
       (await client.checkOrgMembership(orgs)) ||
       client.isExcludedLogin(loginsToIgnore)
     ) {
-      core.info("Looks like this issue doesn't need labeling! 👍")
+      if (failIfMember && !client.hasLabel(labelName)) {
+        core.setFailed(
+          'The PR does not have any labels, please add the labels!'
```
Comment (jordanbreen28) [inline src/main.ts:31]:
```suggestion
          'The PR does not have a label, please add a label!'
```

**Comment by jordanbreen28** [comment]
Will this trigger again and pass when a label is added?
Also think it would be a good idea to have this for all PRs; not just our ones :)

File: src/main.ts (hunk 1 above)
Comment (malikparvez) [reply src/main.ts]: done

**Comment by malikparvez** [comment]
> Will this trigger again and pass when a label is added? Also think it would be a good idea to have this for all PRs if we could check for a label that isn’t the community one; not just our ones :)

yes it will trigger when pr is opened, label is added and label is removed and added the changes to consider if the pr has any label other than the community

**Comment by jordanbreen28** [comment]
> > Will this trigger again and pass when a label is added? Also think it would be a good idea to have this for all PRs if we could check for a label that isn’t the community one; not just our ones :)
> 
> yes it will trigger when pr is opened, label is added and label is removed and added the changes to consider if the pr has any label other than the community

Nice job! Not sure if i'm missing something but.. I can't see the logic that is checking non-org member PRs that do not have any other label than community.
For example, we would want the workflow to fail on a PR like this one, https://github.com/puppetlabs/puppetlabs-kubernetes/pull/619.

File: src/main.ts [hunk 2]
```diff
@@ -11,21 +11,31 @@ export async function run(): Promise<void> {
...
     const token = core.getInput('token', {required: true})
-
     const client = new GitHubClient(token)
-
-    if (client.hasLabel(labelName)) {
-      core.info(`The '${labelName}' label is already present on this issue! 🙌`)
-      return
-    }
+    const labels = client.getLabels()
+    const hasCommunityLabel = labels && !labels.includes(labelName)
```
Comment (jordanbreen28) [inline src/main.ts:19]:
```suggestion
    const missingCommunityLabel = labels && !labels.includes(labelName)
```

File: src/main.ts [hunk 3]
```diff
@@ -11,21 +11,31 @@ export async function run(): Promise<void> {
...
+        )
+      } else {
+        core.info("Looks like this issue doesn't need labeling! 👍")
+      }
+      return
+    }
+
+    if (hasCommunityLabel) {
+      core.setFailed(
+        'Community PR must have a label name community, please add it!'
```
Comment (jordanbreen28) [inline src/main.ts:37]:
```suggestion
        'Community PRs must be labelled!'
```

Comment (malikparvez) [reply src/main.ts:37]: so here community pr's are labelled automatically this error is if the label is anything other than community

Comment (jordanbreen28) [reply src/main.ts:37]: yeah doesnt really make much sense.
`Community PR must have a label name community`

File: src/main.ts [hunk 4]
```diff
@@ -11,21 +11,31 @@ export async function run(): Promise<void> {
...
+    const hasCommunityLabel = labels && !labels.includes(labelName)
 
     if (
       (await client.checkOrgMembership(orgs)) ||
       client.isExcludedLogin(loginsToIgnore)
     ) {
-      core.info("Looks like this issue doesn't need labeling! 👍")
+      if (failIfMember === 'true' && !labels) {
+        core.setFailed(
+          'The PR does not have the required label, please add it!'
```
[the remark this answers is not in the document]
Comment (malikparvez) [reply src/main.ts]: added

File: src/main.ts (hunk 3 above)
Comment (malikparvez) [reply src/main.ts]: done

Comment (malikparvez) [reply src/main.ts]: done

**Review by jordanbreen28 — APPROVED** [review]

**Pull request merged**